components/butterfly-test/src/lib.rs

```diff
@@ -250,24 +249,22 @@ impl SwimNet {
         self.gossip_rounds().iter().map(|r| r + count).collect()
     }
 
-    pub fn check_rounds(&self, rounds_in: &Vec<isize>) -> bool {
-        let mut finished = Vec::with_capacity(rounds_in.len());
-        for (i, round) in rounds_in.into_iter().enumerate() {
-            if self.members[i].paused() {
-                finished.push(true);
-            } else {
-                if self.members[i].swim_rounds() > *round {
-                    finished.push(true);
-                } else {
-                    finished.push(false);
-                }
+    fn check_rounds_impl(
+        &self,
+        rounds_in: &[isize],
+        get_rounds: impl Fn(&Server) -> isize,
+    ) -> bool {
+        for (member, round) in self.members.iter().zip(rounds_in) {
+            if !member.paused() && (get_rounds)(member) <= *round {
+                return false;
             }
         }
-        if finished.iter().all(|m| m == &true) {
-            return true;
-        } else {
-            return false;
-        }
+
+        true
+    }
```

Comment:
`check_rounds ` and `check_gossip_rounds` were identical aside from one identifier, so I couldn't resist.

Additionally, I simplified the logic; there's no need to go through the whole iteration if at the end you're returning that *all* the values were `true`.